rts/Rendering/UniformConstants.h

```diff
@@ -0,0 +1,80 @@
+#ifndef UNIFORM_CONSTANTS_H
+#define UNIFORM_CONSTANTS_H
+
+#include <array>
+#include <memory>
+
+#include "System/float4.h"
+#include "System/Matrix44f.h"
+#include "Rendering/GL/myGL.h"
+#include "Rendering/GL/VBO.h"
+
+struct UniformMatricesBuffer {
+	CMatrix44f screenView;
+	CMatrix44f screenProj;
+	CMatrix44f screenViewProj;
+
+	CMatrix44f cameraView;
+	CMatrix44f cameraProj;
+	CMatrix44f cameraViewProj;
+	CMatrix44f cameraBillboard;
+
+	CMatrix44f cameraViewInv;
+	CMatrix44f cameraProjInv;
+	CMatrix44f cameraViewProjInv;
+
+	CMatrix44f shadowView;
+	CMatrix44f shadowProj;
+	CMatrix44f shadowViewProj;
+
+	//TODO: minimap matrices
+};
+
+struct UniformParamsBuffer {
+	float4 timeInfo; //gameFrame, gameSeconds, drawFrame, frameTimeOffset
+	float4 viewGeometry; //vsx, vsy, vpx, vpy
+	float4 mapSize; //xz, xzPO2
+
+	float4 rndVec3; //new every draw frame. Only xyz are initialized
+};
+
+class UniformConstants {
+public:
+	static UniformConstants& GetInstance() {
+		static UniformConstants uniformConstantsInstance;
+		return uniformConstantsInstance;
+	};
+	static bool Supported() {
+		static bool supported = GLEW_ARB_uniform_buffer_object && GLEW_ARB_shading_language_420pack; //UBO && UBO layout(binding=x)
+		return supported;
+	}
+public:
+	UniformConstants() : umbBufferMap(nullptr), upbBufferMap(nullptr), umbVBO(nullptr), upbVBO(nullptr), umbBufferSize(0), upbBufferSize(0) {};
```

Comment:
`UniformConstants() : umbBufferMap(nullptr), upbBufferMap(nullptr), umbVBO(nullptr), upbVBO(nullptr), umbBufferSize(0), upbBufferSize(0) {};`
->
1) prefer to just initialize these values in the class definition as they don't depend on the constructor parameters
2) further, you don't want to initialize unique_ptr explicitly, it will be nullptr by default
3) and lastly, take care of your initialization order; you should be writing member initialization in the order you've defined them in the class definition, because C++ will actually initialize them in that order (and not the order you've written it), so you might cause a non-obvious bug if there's a dependency between your members